### PR TITLE
#163426571 Set up Empty react 

### PR DESCRIPTION
**Pull request by obasajujoshua31**
#### Description of Task to be completed?

- add webpack configuration for both production and development environment
- add Heroku post build script and Procfile to host app at Heroku
- add babel to transpile at run time

 #### How should this be manually tested?
Clone https://github.com/andela/inception-ah-frontend
CD into inception-ah-frontend 
npm install and npm start
 #### Any background context you want to provide?
you should have nodeJs and git already in your system.
 #### What are the relevant pivotal tracker stories?
[#163426571](https://www.pivotaltracker.com/story/show/163426571)

 #### Screenshots (if appropriate)
screenshot 2019-03-05 at 10 49 14 pm
screenshot 2019-03-05 at 10 49 03 pm

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Ignoring all our node_modules folder
+node_modules/
+configs/
+dist/
+coverage/
```
Comment (Ndifreke) [inline .eslintignore]: Add a new line here

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Dependency Directory
+node_modules
+
+#Build folder 
+dist
+
+# Environment Setting 
+.env
+
+.DS_Store
```
Comment (Ndifreke) [inline .gitignore:10]: Add new line

**Review by Ndifreke — CHANGES_REQUESTED** [review]

**Review by Ndifreke — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: configs/webpack.common.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+const path = require("path");
+
+module.exports = {
+  entry: {
+    bundle: `${appDirectory}/index.js`,
+    vendor: ["react", "react-dom", "react-router-dom"]
+  },
+  output: {
+    path: buildDirectory,
+    filename: "[name].bundle.js"
```
Comment (Ndifreke) [inline configs/webpack.common.js]: Can we build the output into their respective files, I believe if this is possible we will achieve modularity and reduce on javascript size for pages that don't require the entire bundle

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  "root": true,
+  "extends": ["eslint:recommended", "plugin:react/recommended", "airbnb-base"],
+  "env": {
+    "node": true,
+    "es6": true,
+    "browser": true,
+    "jest": true
+  },
+  "globals": {
+    "expect": false
```
Comment (benfluleck) [inline .eslintrc:11]: why are you doing this? You already have jest in your env there is no need for this

File: .eslintrc [hunk 5]
```diff
@@ -0,0 +1,57 @@
+{
+  "root": true,
+  "extends": ["eslint:recommended", "plugin:react/recommended", "airbnb-base"],
+  "env": {
+    "node": true,
+    "es6": true,
+    "browser": true,
```
Comment (benfluleck) [inline .eslintrc:7]: You said no jquery so you shouldn't need Web APis.

File: .eslintrc [hunk 6]
```diff
@@ -0,0 +1,57 @@
+{
+  "root": true,
+  "extends": ["eslint:recommended", "plugin:react/recommended", "airbnb-base"],
+  "env": {
+    "node": true,
+    "es6": true,
```
Comment (benfluleck) [inline .eslintrc:6]: What i this even doing? Are we not writing javascript

File: .postcssrc [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
+ "postcss": {
+   "parser": "sugarss",
+   "map": false,
+   "plugins": {
+     "postcss-plugin": {}
+   }
+ }
+}
```
Comment (benfluleck) [inline .postcssrc:9]: Add a new line here

File: configs/webpack.common.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const { appDirectory, buildDirectory } = require("./filePaths");
+const htmlWebpackPlugin = require("html-webpack-plugin");
+const cleanWebpackPlugin = require("clean-webpack-plugin");
+const path = require("path");
+
+module.exports = {
+  entry: {
+    bundle: `${appDirectory}/index.js`,
+    vendor: ["react", "react-dom", "react-router-dom"]
```
Comment (benfluleck) [inline configs/webpack.common.js]: You don't understand what this line does please go and understand it or remove

File: configs/webpack.common.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+              ]
+            }
+          }
+        ]
+      }
+    ]
+  },
+  resolve: {
+    alias: {
+      "@pages": path.resolve(__dirname, "../src/pages")
```
Comment (benfluleck) [inline configs/webpack.common.js]: This looks like decorator functions use `<pages>`

File: configs/webpack.dev.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+const merge = require("webpack-merge");
+const commonConfig = require("./webpack.common");
+const { buildDirectory } = require("./filePaths");
+
+module.exports = merge(commonConfig, {
+  mode: "development",
+  devServer: {
+    historyApiFallback: true,
+    port: 8000,
+    contentBase: buildDirectory
```
Comment (benfluleck) [inline configs/webpack.dev.js]: This is a repetition

File: configs/webpack.prod.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
+
+const merge = require("webpack-merge");
+const commonConfig = require("./webpack.common");
+const UglifyJsPlugin = require("uglifyjs-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+
+const uglifyJS = new UglifyJsPlugin({
+  test: /\.(js|jsx)$/,
```
Comment (benfluleck) [inline configs/webpack.prod.js]: This is not needed

File: src/components/BaseRoute.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import { Route } from "react-router-dom";
+
+import HomePage from "@pages/Homepage";
```
Comment (benfluleck) [inline src/components/BaseRoute.js]: Change to `<pages>`

File: src/index.scss [hunk 13]
```diff
@@ -0,0 +1,8 @@
+.app {
+  color: orange;
```
Comment (benfluleck) [inline src/index.scss]: Go and meet on a CSS convention

File: src/index.scss [hunk 14]
```diff
@@ -0,0 +1,8 @@
...
+}
+
+* {
```
Comment (benfluleck) [inline src/index.scss:2]: Global styling should be at the top or possibly in another file

File: src/pages/Homepage.js [hunk 15]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from "react";
```
Comment (benfluleck) [inline src/pages/Homepage.js]: Destructure `{Fragment}` from here

**Review by benfluleck — CHANGES_REQUESTED** [review]

**Pull request closed without merging**